src/HtmlGenerator/Pass1-Generation/XmlSupport.cs

```diff
@@ -55,66 +46,63 @@ private string GetHtml(string sourceXmlFilePath, string destinationHtmlFilePath,
 
             var url = "/#" + assemblyName + "/" + displayName.Replace('\\', '/');
 
-            var file = string.Format("File: <a id=\"filePath\" class=\"blueLink\" href=\"{0}\" target=\"_top\">{1}</a><br/>", url, displayName);
-            var row = string.Format("<tr><td>{0}</td></tr>", file);
-            Markup.WriteLinkPanel(s => sb.AppendLine(s), row);
+            Markup.WriteLinkPanel(
+                s => sb.AppendLine(s),
+                fileLink: (displayName, url),
+                webAccessUrl: ProjectGenerator.GetWebAccessUrl(sourceXmlFilePath),
+                projectLink: (ProjectGenerator.ProjectSourcePath, Url: "/#" + assemblyName, assemblyName));
 
             // pass a value larger than 0 to generate line numbers statically at HTML generation time
             var table = Markup.GetTablePrefix();
             sb.AppendLine(table);
 
-            if (sourceText.Length > 1000000)

```

Comment:
Careful, I added this check just yesterday:
https://github.com/KirillOsenkov/SourceBrowser/commit/ad901faf42a0d3e516b7d856f626f427e724eee2

To avoid generating XML for large files because it was causing OOM for source.dot.net (they have giant resource XML files)

Reply:
That is very strange. I did `git pull upstream main --rebase`; maybe I crossed it with an editor save then.